src/client/Client.cc

```diff
@@ -6852,6 +6852,7 @@ void Client::fill_statx(Inode *in, unsigned int mask, struct ceph_statx *stx)
   /* These are always considered to be available */
   stx->stx_dev = in->snapid;
   stx->stx_blksize = MAX(in->layout.stripe_unit, 4096);
+  stx->stx_mode = S_IFMT & in->mode;
```

Comment:
No, we want logical AND. David H.'s original statx set says:

 (2) st_mode.
     The part of this that identifies the file type will always be
     available, irrespective of the setting of STATX_MODE.  The access
     flags and sticky bit are as for class (1).


In our case, the caller needs to check for (or request) CEPH_STATX_MODE if he wants to use the permission bits. If he doesn't have that, then he can still access the type bits, but nothing else in the mode.

In principle, we could just set the full mode unconditionally there and only set CEPH_STATX_MODE flag in stx_mask if we have As caps, but I think it's a little cleaner to avoid setting any of the permission bits if we don't hold As.